app/code/Magento/InventoryCatalog/etc/di.xml

```diff
@@ -96,6 +96,10 @@
         <plugin name="adapt_get_product_stock_status" type="Magento\InventoryCatalog\Plugin\CatalogInventory\Api\StockRegistry\AdaptGetProductStockStatusPlugin"/>
         <plugin name="adapt_get_product_stock_status_by_sku" type="Magento\InventoryCatalog\Plugin\CatalogInventory\Api\StockRegistry\AdaptGetProductStockStatusBySkuPlugin"/>
     </type>
+    <type name="Magento\Catalog\Model\ProductRepository">
+        <plugin name="cleanLocalCacheAfterSaveOrDeleteProduct" type="Magento\InventoryCatalog\Plugin\CatalogInventory\Model\ProductRepository\CleanGetProductIdsBySkusCache"/>
+        <plugin name="cleanLocalProductTypesCacheAfterSaveOrDeleteProduct" type="Magento\InventoryCatalog\Plugin\CatalogInventory\Model\ProductRepository\CleanGetProductTypesBySkusCache"/>
```

Comment:
two more plugins on Product Repository will just lead to additional degradation of performance